hathor/transaction/scripts/execute.py

```diff
@@ -54,21 +54,19 @@ def execute_eval(data: bytes, log: list[str], extras: ScriptExtras) -> None:
         :raises ScriptError: case opcode is not found
         :raises FinalStackInvalid: case the evaluation fails
     """
-    from hathor.transaction.scripts.opcode import MAP_OPCODE_TO_FN, Opcode
+    from hathor.transaction.scripts.opcode import Opcode, execute_op_code
+    from hathor.transaction.scripts.script_context import ScriptContext
+    context = ScriptContext()
     stack: Stack = []
     data_len = len(data)
     pos = 0
     while pos < data_len:
         opcode, pos = get_script_op(pos, data, stack)
         if Opcode.is_pushdata(opcode):
             continue
-        # this is an opcode manipulating the stack
-        fn = MAP_OPCODE_TO_FN.get(opcode, None)
-        if fn is None:
-            # throw error
-            raise ScriptError('unknown opcode')
 
-        fn(stack, log, extras)
+        # this is an opcode manipulating the stack
+        execute_op_code(Opcode(opcode), stack, log, extras, context)
```

Comment:
Maybe log which opcode is being executed?

Reply:
Done in https://github.com/HathorNetwork/hathor-core/pull/851/commits/b8ac57b0f42361e825abdbdcf58fec6e079d70a9 (inside the `execute_op_code` function)

Reply:
Thanks!